src/MongoDB.Analyzer/Core/TypesProcessor.cs

```diff
@@ -22,6 +22,8 @@ internal sealed class TypesProcessor
 
     public MemberDeclarationSyntax[] TypesDeclarations => _processedTypes.Values.Select(p => p.NewDeclaration).ToArray();
 
+    public Dictionary<string, string> GeneratedTypeToOriginalTypeMapping => _processedTypes.ToDictionary(x => x.Value.NewName, x => x.Key);
```

Comment:
Turn it to IEnumerable<(NewName, PreviousName)>, to skip Dictionary object creation.

Reply:
Done